### PR TITLE
perf: Schedule rate-limit notifications on shared executor (JAVA-653)

### PR DESCRIPTION
**Pull request by runningcode**
## :scroll: Description

`RateLimiter` created a `java.util.Timer` (a dedicated thread) whose thread stayed alive for the rest of the process once the SDK got rate limited. The "rate limit lifted" observer notification now runs on the shared timer executor (`SentryOptions#getTimerExecutorService`), already used for transaction timeouts, whose single worker thread is reused and self-terminates when idle.

- The notification is scheduled with the delay until the limit expires; to avoid an extra `getCurrentTimeMillis()` call the already-computed `retryAfterMillis` is passed through as the delay.
- Pending notification futures are tracked (completed ones pruned on each schedule) and cancelled on `close()`, preserving the old `Timer#cancel()` semantics.

## :bulb: Motivation and Context

Part of reducing the number of threads created by the SDK: [JAVA-653](https://linear.app/getsentry/issue/JAVA-653/reduce-number-of-threads-created-and-used-by-the-sdk).

Once an app got rate limited, this timer thread lived forever. The shared timer executor's worker is reused and idles out.

## :green_heart: How did you test it?

Existing `RateLimiterTest`, adapted from the Timer-mock verification to the executor/future model, plus `AsyncHttpTransportTest`.

## :pencil: Checklist

- [x] I added GH Issue ID _&_ Linear ID
- [x] I added tests to verify the changes.
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] I updated the wizard if needed.
- [ ] Review from the native team if needed.
- [x] No breaking change or entry added to the changelog.
- [x] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

## :crystal_ball: Next steps

Related PRs in this effort: LifecycleWatcher (#5819), performance collector (#5816), HostnameCache (#5817), batch processors (#5818).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: sentry/src/test/java/io/sentry/transport/RateLimiterTest.kt [hunk 1]
```diff
@@ -66,6 +66,8 @@ class RateLimiterTest {
 
     fun getSUT(): RateLimiter {
       val options = SentryOptions().apply { setLogger(NoOpLogger.getInstance()) }
+      // a real executor so scheduled rate-limit-lifted notifications actually run
+      options.setTimerExecutorService(SentryExecutorService(options))
```
Comment (romtsn) [inline sentry/src/test/java/io/sentry/transport/RateLimiterTest.kt]: do we need to close it in teardown perhaps, so it doesn't leak across tests?

File: sentry/src/main/java/io/sentry/transport/RateLimiter.java [hunk 2]
```diff
@@ -312,19 +315,25 @@ private void applyRetryAfterOnlyIfLonger(
 
       notifyRateLimitObservers();
 
-      try (final @NotNull ISentryLifecycleToken ignored = timerLock.acquire()) {
-        if (timer == null) {
-          timer = new Timer(true);
+      // notify observers again once the rate limit is lifted, using the shared timer executor
+      // instead of a dedicated Timer thread
+      try (final @NotNull ISentryLifecycleToken ignored = notifyFuturesLock.acquire()) {
```
Comment (romtsn) [inline sentry/src/main/java/io/sentry/transport/RateLimiter.java:320]: `l`: unsure if we need to protect this with checking for `instanceof NoOpSentryExecutorService` to avoid allocating FutureTasks for nothing (`isDone` also returns `false` when no-op, so pruning wouldn't do anything), but we're not really doing that anywhere else I believe, so I'm fine with keeping it as-is.

**Review by romtsn — APPROVED** [review]
couple of nits, but LGTM 🎉

File: sentry/src/main/java/io/sentry/transport/RateLimiter.java (hunk 2 above)
Comment (runningcode) [reply sentry/src/main/java/io/sentry/transport/RateLimiter.java:320]: Yeah it is a good point but also quite an edge case. Since we aren't doing it anywhere else, I don't think we should do it here.

File: sentry/src/test/java/io/sentry/transport/RateLimiterTest.kt (hunk 1 above)
Comment (runningcode) [reply sentry/src/test/java/io/sentry/transport/RateLimiterTest.kt]: Good point!

File: sentry/src/main/java/io/sentry/transport/RateLimiter.java [hunk 3]
```diff
@@ -312,19 +315,25 @@ private void applyRetryAfterOnlyIfLonger(
...
+        while (iterator.hasNext()) {
+          if (iterator.next().isDone()) {
+            iterator.remove();
+          }
+        }
+        try {
+          notifyObserversFutures.add(
+              options
+                  .getTimerExecutorService()
+                  .schedule(this::notifyRateLimitObservers, delayMillis));
```
[the remark this answers is not in the document]
Comment (runningcode) [reply sentry/src/main/java/io/sentry/transport/RateLimiter.java:331]: Created https://github.com/getsentry/sentry-java/pull/5851 to follow up on this
